### PR TITLE
Remove hardcoded text from Ulduar Teleport and Auriaya 

### PR DESCRIPTION
**Pull request by a4501150**
##### CHANGES PROPOSED:

-  see [PR 824](https://github.com/azerothcore/azerothcore-wotlk/pull/824)
-  fixes its typo and broken conditions

###### ISSUES ADDRESSED:

Based on [PR 824](https://github.com/azerothcore/azerothcore-wotlk/pull/824) 

##### TESTS PERFORMED:

Tested in game

##### KNOWN ISSUES AND TODO LIST:

- [x] formation grounds gossip menu item would disappear if you kill the first boss (although you don't need it any more since the death of the first boss unlocks next location's gossip menu)

- [ ] the whole instance of Ulduar needs to be rewritten, everything of text is hardcoded

- [x] teleport spells in AC are broken, thus I can only write spell scripts for them.

##### Target branch(es):

Master

### CODE REVIEW

File: src/server/scripts/Custom/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+LambseaScripts/*
+LambseaScripts
```
[the remark this answers is not in the document]
Comment (a4501150) [reply src/server/scripts/Custom/.gitignore]: deleted

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_xt002.cpp [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/*
+﻿/*
```
Comment (Sdyess) [inline src/server/scripts/Northrend/Ulduar/Ulduar/boss_xt002.cpp]: This file shouldn't be included in this pull request.

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_auriaya.cpp [hunk 3]
```diff
@@ -2,6 +2,7 @@
  * Originally written by Xinef - Copyright (C) 2016+ AzerothCore <www.azerothcore.org>, released under GNU AGPL v3 license: https://github.com/azerothcore/azerothcore-wotlk/blob/master/LICENSE-AGPL3
```
Comment (Sdyess) [inline src/server/scripts/Northrend/Ulduar/Ulduar/boss_auriaya.cpp]: This file shouldn't be included in this pull request.

File: src/server/scripts/Northrend/Ulduar/Ulduar/ulduar.cpp [hunk 4]
```diff
@@ -502,11 +410,98 @@ class go_call_tram : public GameObjectScript
...
+        BASE_CAMP = 64014,
+        GROUNDS = 64032,
+        FORGE = 64028,
+        SCRAPYARD = 64031,
+        ANTECHAMBER = 64030,
+        WALKWAY = 64029,
+        CONSERVATORY = 64024,
+        MADNESS = 64025,
+        SPARK = 65042
+    };
```
Comment (Sdyess) [inline src/server/scripts/Northrend/Ulduar/Ulduar/ulduar.cpp]: I'm okay with this because it at least avoids magic numbers, but maybe better in a header file somewhere if one exists?

File: src/server/scripts/Northrend/Ulduar/Ulduar/ulduar.cpp [hunk 5]
```diff
@@ -502,11 +410,98 @@ class go_call_tram : public GameObjectScript
...
+            }
+
+            return SPELL_CAST_OK;
+        }
+
+        //doing real teleporting because all spells are just broken
+        void TeleportPlayer()
+        {
+            auto SpellID = GetSpellInfo()->Id;
+            auto player = (Player*)GetExplTargetUnit();
```
Comment (Sdyess) [inline src/server/scripts/Northrend/Ulduar/Ulduar/ulduar.cpp]: Don't use a C style cast please. static cast or reinterpret cast, based on whatever your plan is with it. The auto keyword seems weird here, the functions that get the type aren't explicit in what they're returning in terms of type.

See here: http://www.acodersjourney.com/2016/02/c-11-auto/

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_xt002.cpp (hunk 2 above)
Comment (Sdyess) [reply src/server/scripts/Northrend/Ulduar/Ulduar/boss_xt002.cpp]: Yes but it is not relevant to the subject of your PR. This is only about the teleporter.

**Comment by BarbzYHOOL** [comment]
@Viste if he did this it's because he tested your PR and it didn't work (the conditions) and you were absent for months

**Comment by a4501150** [comment]
> > @Viste if he did this it's because he tested your PR and it didn't work (the conditions) and you were absent for months
> 
> sry bro now i can help anything becous of life troubles, when i can i rework my pr to whole ulduar(i hope in December) and do all other what want to do
> btw in sunwell smart script what i use work fine...

if you gonna rework the whole script then I may delete my branch and close this PR

**Comment by Sdyess** [comment]
If someone could test this PR, could probably be merged after a confirmation.

**Comment by BarbzYHOOL** [comment]
There is no information on how to test though

**Comment by Viste** [comment]
@Nefertumm its just copy of my PR with little fixes, and i do update my PR in January after new yr, in update i fix all broken stuff and add more fixes for hardcodes in ulduar

**Comment by Viste** [comment]
update original PR
